### PR TITLE
Dc and GRN issues were handled

### PR DESCRIPTION
**Pull request by KaviyaPeriyasamy**
This PR helps to resolve the below issue,
1) Select Company address automatically in DC
2) Sort the return items table in DC in Dyeing (color ascending -then- dia ascending)
3) Move projected column to left in the items return table in DC
4) Colour-Dia wise ordering as point 2 in GRN
5) Button to delete the zero rows in GRN

### CODE REVIEW

File: apparelo/apparelo/doctype/dc/dc.py [hunk 1]
```diff
@@ -510,8 +512,10 @@ def get_expected_items_in_return(doc, items_to_be_sent=None, use_delivery_qty=Fa
 		if not use_delivery_qty:
 			item_to_be_received['projected_qty'] = item_to_be_received['qty']
 			item_to_be_received['qty'] = 0
-	
-	items_to_be_received = sorted(items_to_be_received, key = lambda i: i['attribute']) 
+	if 'Apparelo Colour' in item_to_be_received and 'Dia' in item_to_be_received:
+		items_to_be_received = sorted(sorted(items_to_be_received, key = lambda i: i['Dia']), key = lambda i: i['Apparelo Colour'])
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/dc/dc.py]: Please handle cases where only 'Dia' is there as an attribute.

File: apparelo/apparelo/doctype/dc/dc.js [hunk 2]
```diff
@@ -4,6 +4,11 @@
 frappe.ui.form.on('DC', {
 	onload: function(frm) {
 		frm.set_value("company",frappe.defaults.get_default("company"));
+		frappe.db.get_list("Address",{filters:{is_primary_address: 1, is_shipping_address: 1}}).then(data => {
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/dc/dc.js]: remove 'is_primary_address' filter...

File: apparelo/apparelo/doctype/dc/dc.js [hunk 3]
```diff
@@ -250,7 +255,11 @@ var update_company_address = function(frm){
 					frm.set_value("company_address_name",r.message)
 				}
 				else {
-					frm.set_value("company_address_name","")
+					frappe.db.get_list("Address",{filters:{is_primary_address: 1, is_shipping_address: 1}}).then(data => {
+						if(data && data.length){
+							frm.set_value("company_address_name",data[0].name);
+						}
+					})
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/dc/dc.js]: check and remove this else if it is not required

File: apparelo/apparelo/doctype/grn/grn.py [hunk 4]
```diff
@@ -115,11 +115,25 @@ def get_items(doc):
...
+				return_material['Apparelo Size'] = attr.attribute_value
 		if apparelo_process:
-			return_materials.append({"item_code":item.item_code,"uom":item.uom,"qty":item.qty,"pf_item_code":item_detail.print_code,"secondary_uom":apparelo_process.in_secondary_uom})
+			return_material.update({"item_code":item.item_code,"uom":item.uom,"qty":item.qty,"pf_item_code":item_detail.print_code,"secondary_uom":apparelo_process.in_secondary_uom})
+			return_materials.append(return_material)
 		else:
-			return_materials.append({"item_code":item.item_code,"uom":item.uom,"qty":item.qty,"pf_item_code":item_detail.print_code,"secondary_uom":item.uom})
+			return_material.update({"item_code":item.item_code,"uom":item.uom,"qty":item.qty,"pf_item_code":item_detail.print_code,"secondary_uom":item.uom})
+			return_materials.append(return_material)
+	if 'Apparelo Colour' in return_materials[0] and 'Dia' in return_materials[0]:
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/grn/grn.py]: handle only 'Dia' cases

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/dc/dc.py (hunk 1 above)
Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/dc/dc.py]: Check this commit https://github.com/aerele/apparelo/pull/182/commits/fad746d2885ad2560b0b0562f331bf9b0246b02c

File: apparelo/apparelo/doctype/dc/dc.js (hunk 2 above)
Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/dc/dc.js]: Resolved by this https://github.com/aerele/apparelo/pull/182/commits/8302d6988855b71cd9f14c0142edd5e928b1c337

**Comment by KaviyaPeriyasamy** [comment]
This commit https://github.com/aerele/apparelo/pull/182/commits/3427f52f08a0a0f0a0f2442c675c461884b7bbca will create non-mandatory ```location``` custom field in the address. Related patches were written and this function called at the time of installation. Removed the old patch file which is not needed.

**Comment by KaviyaPeriyasamy** [comment]
This commit https://github.com/aerele/apparelo/pull/182/commits/fa6ec0711334f2460a7de63a202e4799434a433c will help to create default roles in the time of installation and migration.

File: apparelo/apparelo/doctype/dc/dc.py (hunk 1 above)
Comment (karthikeyan5) [reply apparelo/apparelo/doctype/dc/dc.py]: looks good... cool...

File: apparelo/apparelo/doctype/dc/dc.js (hunk 2 above)
Comment (karthikeyan5) [reply apparelo/apparelo/doctype/dc/dc.js]: 👍🏽

**Review by karthikeyan5 — APPROVED** [review]

**Pull request merged**